src/shared/components/views/DatatableWithToggle.tsx

```diff
@@ -20,29 +25,37 @@ const DatatableWithToggle = ({ table }: { table: JSX.Element }) => {
     'protvista-datatable'
   );
 
-  const toggleTableState = async () => {
-    const tableEl = tableRef?.current;
-    if (tableEl) {
-      if (tableState === 'Expand') {
-        tableEl.expandTable = true;
-        setTableState('Collapse');
-      } else {
-        tableEl.expandTable = false;
-        setTableState('Expand');
-        await frame();
-        tableContainerRef?.current?.scrollIntoView({
+  // On expand/collapse change
+  useEffect(() => {
```

Comment:
Also, since the collapsed state is a react state, we can put it as a dependency of a useEffect, meaning that it will be called after the change has been applied to the DOM by React, removing the hack I had previously suggested involving waiting

Reply:
Great, it makes sense